examples/advanced/llm_hf/code/hf_peft_model.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+from peft import LoraConfig, get_peft_model
+from transformers import AutoModelForCausalLM
+
+
+class CausalLMPEFTModel(torch.nn.Module):
+    def __init__(self, model_path):
+        super(CausalLMPEFTModel, self).__init__()
+        self.model_path = model_path
+        # PEFT configs
+        peft_config = LoraConfig(
```

Comment:
Is only LoRA supported? Can PEFT params be made configurable?

Reply:
yes, configurable for p-tuning, etc. added the information